test/objectwrap.cc

```diff
@@ -3,6 +3,7 @@
 class TestIter : public Napi::ObjectWrap<TestIter> {
 public:
   TestIter(const Napi::CallbackInfo& info) : Napi::ObjectWrap<TestIter>(info) {}
+  ~TestIter() {Constructor.Reset();}
```

Comment:
This looks wrong.  Why is `Constructor` static if you are resetting it in the destructor?  Should it be a data member or should `Initialize()` call `Constructor.SuppressDestruct()`?

Reply:
I suppose it's because Constructor is a box which may be empty, and the box must be emptied before V8 concludes operations. The library destructor gets rid of the box, but we must get rid of its contents sooner than that. We can call `SuppressDestruct()`, but it'd make no difference since we'd have to call `Constructor.Reset()` from `~TestIter()` anyway, lest we leak the reference.

Reply:
Okay, but why isn't `Constructor` a data member in that case?  I can see why it should be either a singleton or a member but not this half-way hybrid.

Couldn't you call `DefineClass()` in `Iter()` and just omit the call to `Persistent()`?